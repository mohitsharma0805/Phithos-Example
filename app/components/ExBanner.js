import React from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import config from '../config';

const ExBanner = () => {
  const getDeviceType = () => {
    let deviceType = DeviceInfo.getModel();
    if (DeviceInfo.isEmulatorSync) {
      if (Platform.OS === 'ios') {
        deviceType = 'Simutator';
      } else {
        deviceType = 'Emulator';
      }
    }
    return deviceType;
  };

  return (
    <View style={styles.contentView}>
      <Text style={styles.text}>{getDeviceType()}</Text>
    </View>
  );
};

export default ExBanner;

const styles = StyleSheet.create({
  contentView: {
    justifyContent: 'center',
    height: 40,
    width: '100%',
    backgroundColor: config.colors.black,
  },
  text: { marginLeft: 20, color: config.colors.white, fontSize: 17 },
});
