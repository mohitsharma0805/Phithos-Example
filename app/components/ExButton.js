import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import config from '../config';

const ExButton = ({ buttonStyle, text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, buttonStyle]}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ExButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 20,
    borderRadius: 5,
    height: 50,
    backgroundColor: config.colors.appTheme,
  },
  text: { color: config.colors.white, fontSize: 17 },
});
