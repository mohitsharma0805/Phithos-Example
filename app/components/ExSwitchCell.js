import React from 'react';
import { View, StyleSheet, Text, Switch } from 'react-native';
import config from '../config';

const ExSwitchCell = ({ switchValue, onSwitchValueChange }) => {
  const switchValueChanged = (value) => onSwitchValueChange?.(value);
  return (
    <View style={styles.container}>
      <View style={styles.wrapper}>
        <View style={styles.textWrapper}>
          <Text style={styles.title}>Banner Status</Text>
        </View>
        <Switch
          style={styles.switch}
          value={switchValue}
          onValueChange={switchValueChanged}
          trackColor={{
            true: config.colors.appTheme,
            false: null,
          }}
        />
      </View>
    </View>
  );
};

export default ExSwitchCell;

const styles = StyleSheet.create({
  container: { backgroundColor: config.colors.white },
  wrapper: { flexDirection: 'row', alignItems: 'center', height: 60 },
  textWrapper: { flex: 1 },
  title: {
    marginLeft: 16,
    paddingVertical: 2,
    fontSize: 17,
  },
  switch: { marginRight: 20, marginLeft: 16 },
});
