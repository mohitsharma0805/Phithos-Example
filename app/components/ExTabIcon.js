import React from 'react';
import { View, StyleSheet } from 'react-native';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

const ExTabIcon = ({ containerStyle, iconName, iconStyle }) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <SimpleLineIcons name={iconName} style={[styles.icon, iconStyle]} />
    </View>
  );
};

export default ExTabIcon;

const styles = StyleSheet.create({
  container: {
    width: 26,
    height: 26,
    margin: 5,
  },
  icon: { marginTop: 4, fontSize: 22 },
});
