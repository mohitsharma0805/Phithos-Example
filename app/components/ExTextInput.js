import React from 'react';
import { View, StyleSheet, TextInput } from 'react-native';
import config from '../config';

const ExTextInput = ({ viewStyle, ...props }) => {
  return (
    <View style={[styles.contentView, viewStyle]}>
      <TextInput {...props} style={styles.textInput} />
      <View style={styles.divider} />
    </View>
  );
};

export default ExTextInput;

const styles = StyleSheet.create({
  contentView: {
    marginLeft: 20,
    marginRight: 20,
    height: 50,
  },
  textInput: {
    flex: 1,
    marginTop: 12,
    height: 30,
    fontSize: 15,
    padding: 5,
  },
  divider: {
    marginTop: 5,
    height: 1,
    backgroundColor: config.colors.cellDivider,
  },
});
