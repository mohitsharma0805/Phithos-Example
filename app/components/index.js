export { default as ExButton } from './ExButton';
export { default as ExTextInput } from './ExTextInput';
export { default as ExBanner } from './ExBanner';
export { default as ExSwitchCell } from './ExSwitchCell';
