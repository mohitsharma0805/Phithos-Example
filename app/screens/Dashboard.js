import React from 'react';
import { View, StyleSheet, SafeAreaView, Text } from 'react-native';
import { useSelector } from 'react-redux';
import { ExBanner } from '../components';
import { selectBannerStatus } from '../state/Settings';
import { selectUser } from '../state/User';

const Dashboard = () => {
  const bannerStatus = useSelector(selectBannerStatus);
  const user = useSelector(selectUser);

  return (
    <SafeAreaView style={styles.contentView}>
      {bannerStatus && <ExBanner />}
      <View style={styles.container}>
        <Text style={styles.text}>{user.email}</Text>
      </View>
    </SafeAreaView>
  );
};

export default Dashboard;

const styles = StyleSheet.create({
  contentView: { flex: 1 },
  container: { flex: 1, justifyContent: 'center', alignItems: 'center' },
  text: { fontSize: 22 },
});
