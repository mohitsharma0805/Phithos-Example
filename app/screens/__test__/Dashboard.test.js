import React from 'react';
import { shallow } from 'enzyme';
import { SafeAreaView, View, Text } from 'react-native';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';
import Dashboard from '../Dashboard';
import { ExBanner } from '../../components';

describe('Dashboard', () => {
  describe('Render UI with show banner', () => {
    const sagaMiddleware = createSagaMiddleware();
    const mockStore = configureStore([sagaMiddleware]);
    const store = mockStore({
      banner: true,
      user: { email: 'mohit@gmail.com' },
    });

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => store.getState().banner)
      .mockImplementationOnce(() => store.getState().user);

    const wrapper = shallow(<Dashboard />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(View).length).toEqual(1);
      expect(wrapper.find(Text).length).toEqual(1);
      expect(wrapper.find(ExBanner).length).toEqual(1);
    });
  });
  describe('Render UI with banner hidden', () => {
    const sagaMiddleware = createSagaMiddleware();
    const mockStore = configureStore([sagaMiddleware]);
    const store = mockStore({
      banner: false,
      user: { email: 'mohit@gmail.com' },
    });

    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => store.getState().banner)
      .mockImplementationOnce(() => store.getState().user);

    const wrapper = shallow(<Dashboard />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(View).length).toEqual(1);
      expect(wrapper.find(Text).length).toEqual(1);
    });
  });
});
