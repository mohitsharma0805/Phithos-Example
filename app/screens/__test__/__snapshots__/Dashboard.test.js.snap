// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Das UI test cases renders the main view 1`] = `ShallowWrapper {}`;

exports[`Dashboard Render UI with banner hidden renders the main view 1`] = `ShallowWrapper {}`;

exports[`Dashboard Render UI with show banner renders the main view 1`] = `ShallowWrapper {}`;

exports[`Dashboard UI test cases renders the main view 1`] = `ShallowWrapper {}`;
